src/types/mod.rs

```diff
@@ -50,24 +60,43 @@ pub enum BlockStatus {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct BlockDetails {
+    /// Number of the block.
     pub number: u64,
+    /// Corresponding L1 batch number.
     pub l1_batch_number: u64,
+    /// Address of the operator who committed the block.
     pub operator_address: Address,
+    /// Version of the ZKsync protocol the block was committed under.
     pub protocol_version: Option<String>,
+    /// Unix timestamp when the block was committed.
     pub timestamp: u64,
+    /// Number of L1 transactions included in the block.
     pub l1_tx_count: u64,
+    /// Number of L2 transactions included in the block.
     pub l2_tx_count: u64,
+    /// Root hash of the block's state after execution.
     pub root_hash: Option<B256>,
+    /// Current status of the block: verified or sealed.
     pub status: BlockStatus,
+    /// Transaction hash of the commit operation on L1.
     pub commit_tx_hash: Option<B256>,
+    /// Timestamp when the block was committed on L1.
     pub committed_at: Option<DateTime<Utc>>,
+    /// Transaction hash of the proof submission on L1.
     pub prove_tx_hash: Option<B256>,
+    /// Timestamp when the proof was submitted on L1.
     pub proven_at: Option<DateTime<Utc>>,
+    /// Transaction hash of the execution on L1.
     pub execute_tx_hash: Option<B256>,
+    /// Timestamp when the block execution was completed on L1.
     pub executed_at: Option<DateTime<Utc>>,
```

Comment:
Probably makes sense to mention something like `for the batch containing this L2 block`.